src/plugins/intel_cpu/src/transformations/snippets/aarch64/pass/brgemm_to_gemm_cpu.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "brgemm_to_gemm_cpu.hpp"
+
+#include "cpu_shape.h"
+#include "openvino/core/graph_util.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/pass/pattern/matcher.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "snippets/itt.hpp"
+#include "snippets/op/brgemm.hpp"
+#include "snippets/op/buffer.hpp"
+#include "snippets/utils/utils.hpp"
+#include "transformations/snippets/aarch64/op/gemm_copy_b.hpp"
+#include "transformations/snippets/aarch64/op/gemm_cpu.hpp"
+#include "transformations/tpp/common/op/modifiers.hpp"
+#include "utils/general_utils.h"
+
+namespace ov::intel_cpu {
+
+using namespace snippets::lowered;
+
+namespace {
+template <typename T>
+void set_full_port_desc(const T& port) {
+    const auto& shape_rank = port.get_partial_shape().size();
+    const std::vector<size_t> full_dim_subtensor(std::min(shape_rank, static_cast<size_t>(2)),
+                                                 ov::snippets::utils::get_full_dim_value());
+    PortDescriptorUtils::set_port_descriptor(port, full_dim_subtensor);
+}
+}  // namespace
+
+pass::BrgemmToGemmCPU::BrgemmToGemmCPU() {
+    MATCHER_SCOPE(BrgemmToGemmCPU);
+    auto is_not_tpp = [](const Output<Node>& out) {
+        return !std::dynamic_pointer_cast<const intel_cpu::tpp::modifier::TensorProcessingPrimitive>(
+            out.get_node_shared_ptr());
+    };
+    auto m_brgemm = ov::pass::pattern::wrap_type<snippets::op::Brgemm>(is_not_tpp);
+
+    auto callback = [=](ov::pass::pattern::Matcher& m) {
+        OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "ov::intel_cpu::pass::BrgemmToGemmCPU")
+        const auto node = m.get_match_root();
+        const auto brgemm = ov::as_type_ptr<snippets::op::Brgemm>(node);
+        const auto brgemm_plugin = ov::as_type_ptr<aarch64::GemmCPU>(node);
+        if (!brgemm || brgemm_plugin) {
+            OPENVINO_THROW("GemmCPU cannot be in body before BrgemmToGemmCPU pass");
+        }
```

Comment:
Minor: can be replaced with OPENVINO_ASSERT

Reply:
done, Thanks Vladislav!